src/services/GetIdProducts.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { Wine } from '../types/IProducts';
+
+const useGetProductById = () => {
+    const [data, setData] = useState<Wine[]>([]);
+
+    useEffect(() => {
+        const localStorageId = localStorage.getItem("id");
+
+        async function getProductId() {
+            try {
+                const response = await axios.get("http://localhost:3000/products");
```

Comment:
You're making this request to get only one product. Also, this same request is being made in home page. What you think about store this returned data in a shared place throughout the application? You can use React Hooks to do this.